### ISSUE TITLE
Use maskrcnn to train my own dataset

### ISSUE THREAD

**Issue by dorest** [dmlc/gluon-cv#380]
I made my own dataset from the coco dataset and verified that all the pictures in the set were trained pictures. But I picked out 10 of them and found that after using the validation steps of pycocotools, map was 0. I've only trained two classes, modifying the CLASSES entry in the instances. py file, and if I want to train my dataset I wonder if I've missed anything.

Namespace(batch_size=2, dataset='coco', epochs=26, gpus='0,1', log_interval=100, lr=0.0025, lr_decay=0.1, lr_decay_epoch='17,23', lr_warmup=4000.0, momentum=0.9, network='resnet50_v1b', num_workers=4, resume='', save_interval=1, save_prefix='mask_rcnn_resnet50_v1b_coco', seed=233, start_epoch=0, val_interval=1, verbose=False, wd=0.0001)
Start training from [Epoch 0]
[Epoch 0 Iteration 0] Set learning rate to 0.0008333333333333333
[Epoch 0][Batch 99], Speed: 4.501 samples/sec, RPN_Conf=0.670,RPN_SmoothL1=0.122,RCNN_CrossEntropy=1.202,RCNN_SmoothL1=0.289,RCNN_Mask=0.705,RPNAcc=0.814,RPNL1Loss=0.529,RCNNAcc=0.822,RCNNL1Loss=1.918,MaskAcc=0.539,MaskFGAcc=0.571
[Epoch 0 Iteration 100] Set learning rate to 0.0008749999999999999
Namespace(batch_size=2, dataset='coco', epochs=10, gpus='0,1', log_interval=1, lr=0.0025, lr_decay=0.1, lr_decay_epoch='17,23', lr_warmup=4000.0, momentum=0.9, network='resnet50_v1b', num_workers=4, resume='', save_interval=1, save_prefix='mask_rcnn_resnet50_v1b_coco', seed=233, start_epoch=0, val_interval=1, verbose=False, wd=0.0001)
Start training from [Epoch 0]
[Epoch 0 Iteration 0] Set learning rate to 0.0008333333333333333
[Epoch 0][Batch 0], Speed: 0.855 samples/sec, RPN_Conf=0.689,RPN_SmoothL1=0.173,RCNN_CrossEntropy=4.476,RCNN_SmoothL1=0.187,RCNN_Mask=0.704,RPNAcc=0.641,RPNL1Loss=0.625,RCNNAcc=0.000,RCNNL1Loss=2.660,MaskAcc=0.507,MaskFGAcc=0.553
[Epoch 0 Iteration 1] Set learning rate to 0.00083375
[Epoch 0][Batch 1], Speed: 4.777 samples/sec, RPN_Conf=0.689,RPN_SmoothL1=0.157,RCNN_CrossEntropy=4.370,RCNN_SmoothL1=0.263,RCNN_Mask=0.696,RPNAcc=0.638,RPNL1Loss=0.584,RCNNAcc=0.008,RCNNL1Loss=2.751,MaskAcc=0.522,MaskFGAcc=0.555
[Epoch 0 Iteration 2] Set learning rate to 0.0008341666666666667
[Epoch 0][Batch 2], Speed: 5.145 samples/sec, RPN_Conf=0.689,RPN_SmoothL1=0.160,RCNN_CrossEntropy=4.237,RCNN_SmoothL1=0.251,RCNN_Mask=0.712,RPNAcc=0.662,RPNL1Loss=0.575,RCNNAcc=0.210,RCNNL1Loss=2.575,MaskAcc=0.502,MaskFGAcc=0.553
[Epoch 0 Iteration 3] Set learning rate to 0.0008345833333333331
[Epoch 0][Batch 3], Speed: 5.396 samples/sec, RPN_Conf=0.689,RPN_SmoothL1=0.184,RCNN_CrossEntropy=4.077,RCNN_SmoothL1=0.246,RCNN_Mask=0.714,RPNAcc=0.686,RPNL1Loss=0.564,RCNNAcc=0.378,RCNNL1Loss=2.571,MaskAcc=0.495,MaskFGAcc=0.552
[Epoch 0 Iteration 4] Set learning rate to 0.0008349999999999999
[Epoch 0][Batch 4], Speed: 5.637 samples/sec, RPN_Conf=0.689,RPN_SmoothL1=0.162,RCNN_CrossEntropy=3.872,RCNN_SmoothL1=0.253,RCNN_Mask=0.710,RPNAcc=0.669,RPNL1Loss=0.555,RCNNAcc=0.480,RCNNL1Loss=2.569,MaskAcc=0.503,MaskFGAcc=0.554
[Epoch 0 Iteration 5] Set learning rate to 0.0008354166666666666
[Epoch 0][Batch 5], Speed: 5.417 samples/sec, RPN_Conf=0.688,RPN_SmoothL1=0.177,RCNN_CrossEntropy=3.611,RCNN_SmoothL1=0.279,RCNN_Mask=0.697,RPNAcc=0.687,RPNL1Loss=0.544,RCNNAcc=0.540,RCNNL1Loss=2.567,MaskAcc=0.543,MaskFGAcc=0.554
[Epoch 0 Iteration 6] Set learning rate to 0.0008358333333333333
[Epoch 0][Batch 6], Speed: 5.414 samples/sec, RPN_Conf=0.688,RPN_SmoothL1=0.195,RCNN_CrossEntropy=3.323,RCNN_SmoothL1=0.285,RCNN_Mask=0.692,RPNAcc=0.697,RPNL1Loss=0.559,RCNNAcc=0.586,RCNNL1Loss=2.516,MaskAcc=0.553,MaskFGAcc=0.555
[Epoch 0] Training cost: 4.616, RPN_Conf=0.688,RPN_SmoothL1=0.195,RCNN_CrossEntropy=3.323,RCNN_SmoothL1=0.285,RCNN_Mask=0.692
[Epoch 0] Validation: 
~~~~ Summary bbox metrics ~~~~
=Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
~~~~ Mean AP for bbox ~~~~
=0.0
~~~~ Summary segm metrics ~~~~
=Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
~~~~ Mean AP for segm ~~~~
=0.0
[Epoch 0] Saving parameters to mask_rcnn_resnet50_v1b_coco_0000_0.0000.params
[Epoch 1][Batch 0], Speed: 0.070 samples/sec, RPN_Conf=0.687,RPN_SmoothL1=0.056,RCNN_CrossEntropy=1.108,RCNN_SmoothL1=0.332,RCNN_Mask=0.658,RPNAcc=0.692,RPNL1Loss=0.543,RCNNAcc=0.621,RCNNL1Loss=2.514,MaskAcc=0.563,MaskFGAcc=0.555
[Epoch 1][Batch 1], Speed: 4.587 samples/sec, RPN_Conf=0.686,RPN_SmoothL1=0.147,RCNN_CrossEntropy=1.345,RCNN_SmoothL1=0.375,RCNN_Mask=0.625,RPNAcc=0.698,RPNL1Loss=0.532,RCNNAcc=0.644,RCNNL1Loss=2.492,MaskAcc=0.583,MaskFGAcc=0.556
[Epoch 1][Batch 2], Speed: 1.197 samples/sec, RPN_Conf=0.687,RPN_SmoothL1=0.104,RCNN_CrossEntropy=1.171,RCNN_SmoothL1=0.299,RCNN_Mask=0.659,RPNAcc=0.691,RPNL1Loss=0.526,RCNNAcc=0.672,RCNNL1Loss=2.467,MaskAcc=0.580,MaskFGAcc=0.556
[Epoch 1][Batch 3], Speed: 4.742 samples/sec, RPN_Conf=0.686,RPN_SmoothL1=0.107,RCNN_CrossEntropy=1.127,RCNN_SmoothL1=0.264,RCNN_Mask=0.615,RPNAcc=0.696,RPNL1Loss=0.519,RCNNAcc=0.695,RCNNL1Loss=2.433,MaskAcc=0.593,MaskFGAcc=0.556
[Epoch 1][Batch 4], Speed: 4.997 samples/sec, RPN_Conf=0.685,RPN_SmoothL1=0.123,RCNN_CrossEntropy=1.227,RCNN_SmoothL1=0.278,RCNN_Mask=0.604,RPNAcc=0.697,RPNL1Loss=0.519,RCNNAcc=0.708,RCNNL1Loss=2.413,MaskAcc=0.606,MaskFGAcc=0.556
[Epoch 1][Batch 5], Speed: 4.830 samples/sec, RPN_Conf=0.685,RPN_SmoothL1=0.120,RCNN_CrossEntropy=1.197,RCNN_SmoothL1=0.274,RCNN_Mask=0.588,RPNAcc=0.702,RPNL1Loss=0.513,RCNNAcc=0.722,RCNNL1Loss=2.412,MaskAcc=0.617,MaskFGAcc=0.557
[Epoch 1][Batch 6], Speed: 5.058 samples/sec, RPN_Conf=0.683,RPN_SmoothL1=0.143,RCNN_CrossEntropy=1.216,RCNN_SmoothL1=0.284,RCNN_Mask=0.580,RPNAcc=0.712,RPNL1Loss=0.518,RCNNAcc=0.730,RCNNL1Loss=2.388,MaskAcc=0.628,MaskFGAcc=0.557
[Epoch 1][Batch 7], Speed: 5.167 samples/sec, RPN_Conf=0.682,RPN_SmoothL1=0.148,RCNN_CrossEntropy=1.201,RCNN_SmoothL1=0.287,RCNN_Mask=0.569,RPNAcc=0.722,RPNL1Loss=0.504,RCNNAcc=0.739,RCNNL1Loss=2.368,MaskAcc=0.642,MaskFGAcc=0.557
[Epoch 1] Training cost: 32.573, RPN_Conf=0.682,RPN_SmoothL1=0.148,RCNN_CrossEntropy=1.201,RCNN_SmoothL1=0.287,RCNN_Mask=0.569

**Comment by dorest** [comment]
I tested the official coco dataset, and map is normal.

**Comment by dorest** [comment]
@zhreshold 
I've solved this problem. When using high resolution images (such as 2448x3264), the size of the image you predicted doesn't match the original image and the call cocoapi fails.
The scaling operation reduces the precision, specifically in the 281st row of the train_mask_rcnn.py file.
im_height, im_width = int (round (im_height / im_scale)), int (round (im_width / im_scale))

**Comment by chrisluu** [comment]
@dorest Hi, I want to ask how to construct your own dataset like coco style?

**Comment by dont32** [comment]
@dorest How solved that problem ?. On my custom dataset  map=0
